### PR TITLE
WIP groups: general refactoring

### PR DESCRIPTION
**Pull request by bouzlibop**
rebased on top of https://github.com/inveniosoftware/invenio/pull/3006

### CODE REVIEW

File: invenio/modules/groups/api.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+from invenio.base.i18n import _
+from invenio.ext.sqlalchemy import db
+from invenio.ext.sqlalchemy.utils import session_manager
+from invenio.modules.accounts.errors import AccountSecurityError, \
+    IntegrityUsergroupError
+from invenio.modules.accounts.models import User, UserUsergroup, Usergroup, \
+    get_groups_user_not_joined as models_get_groups_user_not_joined
+
+
+class GroupsAPI:
```
Comment (jirikuncar) [inline invenio/modules/groups/api.py]: I would call it only `Group`. It's clear that it's THE api from its location.

File: invenio/modules/groups/restful.py [hunk 2]
```diff
@@ -0,0 +1,380 @@
+# -*- coding: utf-8 -*-
+##
```
Comment (jirikuncar) [inline invenio/modules/groups/restful.py]: `s/##/#/`

**Comment by jirikuncar** [comment]
Can we separate changes from `invenio/ext/sqlalchemy/utils.py` to separate PR for fast integration?

**Comment by bouzlibop** [comment]
Work will be moved to https://github.com/inveniosoftware/invenio-groups

**Pull request closed without merging**